src/hsm/PSFCorr.cpp

```diff
@@ -247,18 +253,33 @@ namespace hsm {
         // call find_ellipmom_2
         results.image_bounds = object_image.getBounds();
         try {
-            dbg<<"About to get moments using find_ellipmom_2"<<std::endl;
-            find_ellipmom_2(masked_object_image_cview, amp, results.moments_centroid.x,
-                            results.moments_centroid.y, m_xx, m_xy, m_yy, results.moments_rho4,
-                            precision, results.moments_n_iter, hsmparams);
-            dbg<<"Repackaging find_ellipmom_2 results"<<std::endl;
-
-            // repackage outputs from find_ellipmom_2 to the output CppShapeData struct
-            results.moments_amp = 2.0*amp;
-            results.moments_sigma = std::pow(m_xx*m_yy-m_xy*m_xy, 0.25);
-            results.observed_e1 = (m_xx-m_yy) / (m_xx+m_yy);
-            results.observed_e2 = 2.*m_xy / (m_xx+m_yy);
-            results.moments_status = 0;
+             if (!round_moments) {
+                  dbg<<"About to get moments using find_ellipmom_2"<<std::endl;
+                  find_ellipmom_2(masked_object_image_cview, amp, results.moments_centroid.x,
+                                  results.moments_centroid.y, m_xx, m_xy, m_yy, results.moments_rho4,
+                                  precision, results.moments_n_iter, hsmparams);
+                  dbg<<"Repackaging find_ellipmom_2 results"<<std::endl;
+
+                  // repackage outputs from find_ellipmom_2 to the output CppShapeData struct
+                  results.moments_amp = 2.0*amp;
+                  results.moments_sigma = std::pow(m_xx*m_yy-m_xy*m_xy, 0.25);
+                  results.observed_e1 = (m_xx-m_yy) / (m_xx+m_yy);
+                  results.observed_e2 = 2.*m_xy / (m_xx+m_yy);
+                  results.moments_status = 0;
+             } else {
+                  dbg<<"About to get moments using find_mom_2"<<std::endl;
+                  tmv::Matrix<double> moments(3,3);
+                  double sig = guess_sig;
+                  find_mom_2(masked_object_image_cview, moments, 2,
+                             results.moments_centroid.x, results.moments_centroid.y, sig,
+                             hsmparams->convergence_threshold, results.moments_n_iter, hsmparams);
+                  dbg<<"Repackaging find_mom_2 results"<<std::endl;
+                  results.moments_amp =  3.544907701811*sig*moments(0,0);
```

Comment:
Maybe refer to where these magic numbers / formulae come from?  (Point to later in the code in this file.)

Reply:
Where do they come from?  There is no documentation in the later part of the code where they are used.

Reply:
Hmmm, I thought I found their origin in Hirata & Seljak (2003), based on some math in Bernstein & Jarvis (2002), but the actual equations in those papers seem to point to a slightly different answer.  But then I realized it's because some of the equations use different basis functions for the moments (cartesian vs. polar) than others - see comments starting around line 1357, which describe manipulations of `find_mom_1` outputs, for what made me realize this.  Bottom line is I think I can put it all together into an informative comment if I work carefully on connecting but I don't have time to do it today.  Are you OK with a slight delay?

Reply:
I'm not too concerned about where they come from, but it would be good to have in the code at some point.  The delay is fine if you want to this soon.

Reply:
Well, I have very good intentions but not necessarily the best follow-through at this time of the semester (teaching obligations etc.).  Since it's a matter of comments, let me suggest you not delay this PR for me.  If it's otherwise ready to merge, then we can do so and open an issue to remind me to improve the documentation of these numbers later.